sambacc/ctdb.py

```diff
@@ -483,6 +483,26 @@ def _node_update(cmeta: ClusterMeta, real_path: str) -> bool:
     return True
 
 
+def cluster_meta_to_nodes(cmeta: ClusterMeta, real_path: str) -> None:
+    """Write a nodes file based on the current content of the cluster
+    metadata."""
+    with cmeta.open(locked=True) as cmo:
+        json_data = cmo.load()
+        nodes = json_data.get("nodes", [])
+        _logger.info("Found node metadata: %r", nodes)
+        pnn_max = max(n["pnn"] for n in nodes) + 1  # pnn is zero indexed
+        ctdb_nodes: list[str] = [""] * pnn_max
+        for entry in nodes:
+            pnn = entry["pnn"]
+            expected_line = _entry_to_node(ctdb_nodes, entry)
+            ctdb_nodes[pnn] = expected_line
+        _logger.info("Will write nodes: %s", ctdb_nodes)
+        with open(real_path, "w") as nffh:
+            write_nodes_file(nffh, ctdb_nodes)
+            nffh.flush()
+            os.fsync(nffh)
```

Comment:
Could you please remind me why we need this pattern of doing both `flush` and `os.fsync` ? 
It is repeated in this module; maybe have a local helper-function with minimal doc-string.

Reply:
To be fair it is a copy-paste that was tweaked. The goal of the pattern is to call `flush` to make sure any buffers internal to the process get flushed to the OS. Then `fsync` tells the os to flush any buffers it has and ensure the entire file is written to the storage device. Normally, flush is not required as it is done also by close, but to use fync we need an fd - so we end up having to manually flush first.

Reply:
Fair enough. I still think it would be nice to have it local helper-function but this can wait to future cleanups PR.

Reply:
Yeah, if you note I'm not writing very many tests for some of this stuff either. Normally, I'd criticize myself for that but I'm doing a bunch of prototyping simultaneously on the ceph codebase and sambacc. I'm trying to get as many of these into the codebase here in sambacc so I don't end up with massive private branches on both sides. :-)
I'd welcome anyone to jump in and help contribute tests and cleanups on top. :-)